### PR TITLE
MNTOR-1042 - set up GA4 with GTM for Monitor V2

### PR DESCRIPTION
**Pull request by rhelmer**
# References: 
Jira: MNTOR-1042

# Description

Add Google Analytics (GA4) for Monitor V2

# Screenshot (if applicable)

Not applicable.

# How to test

# Checklist (Definition of Done)
- [x] Localization strings (if needed) have been added.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] I've added or updated the relevant sections in readme and/or code comments
- [ ] I've added a unit test to test for potential regressions of this bug.
- [x] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [x] All acceptance criteria are met.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: .env-dist [hunk 1]
```diff
@@ -100,4 +100,7 @@ EDUCATION_VIDEO_URL_VPN=https://monitor.cdn.mozilla.net/videos/Mozilla_VPN.mp4
 ADMINS=
 
 # Enable monthly cron-job, currently for sending unresolved breach reminder emails
-MONTHLY_CRON_ENABLED=
\ No newline at end of file
+MONTHLY_CRON_ENABLED=
+
+SERVER_GENERATED_NONCE=test123
```
Comment (rhelmer) [inline .env-dist:105]: NOTE - this is just for testing, we need to generate this on every request. I'll fix this before moving this PR out of draft mode.

File: src/app.js [hunk 2]
```diff
@@ -2,6 +2,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+import crypto from 'crypto'
```
Comment (toufali) [inline src/app.js:5]: suggestion: is `crypto` a native node package?  If so, I've found the following convention helpful in identifying when a package originates from node:

```suggestion
import crypto from 'node:crypto'
```

File: src/app.js [hunk 3]
```diff
@@ -61,14 +62,30 @@ if (AppConstants.FXA_ENABLED) {
 }
 
 // disable forced https to allow localhost on Safari
-app.use(
+// Support GA4 per https://developers.google.com/tag-platform/tag-manager/web/csp
+app.use((_req, res, _next) => {
+  res.locals.nonce = crypto.randomBytes(16).toString('hex')
```
Comment (toufali) [inline src/app.js:67]: nice.  This nonce doesn't seem so bad with native crypto support

File: src/controllers/breaches.js [hunk 4]
```diff
@@ -20,7 +20,8 @@ async function breachesPage (req, res) {
     emailCount,
     partial: breaches,
     csrfToken: generateToken(res),
-    fxaProfile: req.user.fxa_profile_json
+    fxaProfile: req.user.fxa_profile_json,
+    nonce: res.locals.nonce
```
Comment (toufali) [inline src/controllers/breaches.js:24]: Ugh, I see where the nonce gets nasty – having to pass it on every page.  If this bothers you, we could consider (under separate ticket) building a middleware or something that automatically passes a data object to main on every request, without having to explicitly add it.

On the other hand, sometimes it's good to be explicit 🤷

**Review by toufali — APPROVED** [review]

File: src/app.js [hunk 5]
```diff
@@ -61,14 +62,30 @@ if (AppConstants.FXA_ENABLED) {
...
-      imgSrc,
-      upgradeInsecureRequests: isDev ? null : []
+      upgradeInsecureRequests: isDev ? null : [],
+      scriptSrc: [
+        "'self'",
+        `'nonce-${res.locals.nonce}'`
+      ],
+      imgSrc: [
+        "'self'",
+        'profile.stage.mozaws.net',
```
Comment (rhelmer) [inline src/app.js:77]: Oh I see why this was working before - `imgSrc` is created above this and this is handled there, I'll fix this up.

**Review by mansaj — APPROVED** [review]

**Pull request merged**